lib/main.py

```diff
@@ -85,73 +94,74 @@ def get_file(nc, task_id, file_id):
 LAST_MODEL_NAME = None
 LAST_MODEL = None
 
-class BackgroundProcessTask(threading.Thread):
-    def run(self, *args, **kwargs):  # pylint: disable=unused-argument
-        global ENABLED_FLAG
-        global LAST_MODEL_NAME
-        global LAST_MODEL
-
-        nc = NextcloudApp()
-        while True:
-            if not ENABLED_FLAG:
-                sleep(30)
-                ENABLED_FLAG = nc.enabled_state
-                continue
+def start_bg_task():
+    t = threading.Thread(target=background_thread_task)
+    t.start()
 
-            try:
-                next = nc.providers.task_processing.next_task([f'stt_whisper2:{model_name}' for model_name, _ in models.items()], ['core:audio2text'])
-                if not 'task' in next or next is None:
-                    sleep(5)
-                    continue
-                task = next.get('task')
-            except Exception as e:
-                LOGGER.error(str(e))
-                sleep(5)
+def background_thread_task():
+    global ENABLED
+    global LAST_MODEL_NAME
+    global LAST_MODEL
+
+    nc = NextcloudApp()
+    while True:
+        while not ENABLED.is_set():
+            sleep(5)
+
+        try:
+            next = nc.providers.task_processing.next_task([f'stt_whisper2:{model_name}' for model_name, _ in models.items()], ['core:audio2text'])
+            if not 'task' in next or next is None:
+                wait_for_task()
                 continue
+            task = next.get('task')
+        except Exception as e:
+            LOGGER.error(str(e) + "\n" + "".join(traceback.format_exception(e)))
+            wait_for_task(10)
+            continue
+        try:
+            LOGGER.info(f"Next task: {task['id']}")
+            model_name = next.get("provider").get('name').split(':', 2)[1]
+            LOGGER.info( f"model: {model_name}")
+            if LAST_MODEL_NAME == model_name:
+                model = LAST_MODEL
+            else:
+                model_load = models.get(model_name)
+                if model_load is None:
+                    nc.providers.task_processing.report_result(
+                        task["id"], None, "Requested model is not available"
+                    )
+                    continue
+                model = model_load()
+                LAST_MODEL_NAME = model_name
+                LAST_MODEL = model
+
+            LOGGER.info("generating transcription")
+            time_start = perf_counter()
+            file_name = get_file(nc, task["id"], task.get("input").get('input'))
+            segments, _ = model.transcribe(file_name)
+            del model
+            LOGGER.info(f"transcription generated: {perf_counter() - time_start}s")
+
+            transcript = ''
+            for segment in segments:
+                transcript += segment.text
+            nc.providers.task_processing.report_result(
+                task["id"],
+                {'output': str(transcript)},
+            )
```

Comment:
This is the call that gets stuck :/

Reply:
It seems I was wrong. It was never stuck. It was just running on CPU instead of GPU, which takes ages.... :thinking: 

Works like a charm now. Thanks for your help, though!